frameworks/elastic/docs/tls.md

```diff
@@ -0,0 +1,139 @@
+
+## Install the Enterprise DC/OS CLI
+
+```
+dcos package install --yes dcos-enterprise-cli
+```
+
+## Create key pair
+
+This generates `elastic.private.pem` and `elastic.public.pem`.
+
+```
+dcos security org service-accounts keypair elastic.private.pem elastic.public.pem
+```
+
+## Create service account
+
+The service account name that is configured for your service (`service.service_account` in the service JSON
+configuration or the `service account` text field in the "Edit Configuration" UI) must match the actual service account
+name, which we're calling `elastic` in this case.
+
+```
+dcos security org service-accounts create -p elastic.public.pem -d "Elastic service account" elastic
+```
+
+## Create service account secret
+
+This stores the private key as a [DC/OS secret](https://docs.mesosphere.com/1.10/security/ent/secrets/) named `elastic-secret`.
+
+```
+dcos security secrets create-sa-secret elastic.private.pem elastic elastic-secret
+```
+
+## Add service account to `superusers` group
+
+```
+dcos security org groups add_user superusers elastic
+```
+
+## Install or update Elastic with TLS enabled
+
+Elasticsearch
+[requires that X-Pack is installed](https://www.elastic.co/guide/en/elasticsearch/reference/current/configuring-tls.html)
+to enable transport encryption. Make sure `elasticsearch.xpack_enabled` is set to `true` or nodes won't be able to
+start.
+
+### Install
+
+Make sure to replace `$your_service_name` by your actual service name.
+
+```bash
+cat <<EOF > elastic_with_tls.json
+{
+  "service": {
+    "name": "$your_service_name",
+    "service_account": "elastic",
+    "service_account_secret": "elastic-secret",
+    "security": {
+      "transport_encryption": {
+        "enabled": true
+      }
+    }
+  },
+  "elasticsearch": {
+    "xpack_enabled": true
+  }
+}
+EOF
+```
+
+```
+dcos package install --yes --options=elastic_with_tls.json beta-elastic
+```
+
+### Or update a running service with TLS turned off
+
+Updating requires a `parallel` strategy due to a know
+[Elasticsearch limitation](https://github.com/mesosphere/dcos-commons/blob/e4b2ab5d82a6cda31815fc1224d1eca768513aa9/frameworks/elastic/docs/limitations.md#toggling-tls-requires-doing-a-full-cluster-restart).
+
+```bash
+cat <<EOF > elastic_with_tls.json
+{
+  "service": {
+    "name": "$your_service_name",
+    "service_account": "elastic",
+    "service_account_secret": "elastic-secret",
+    "update_strategy": "parallel",
+    "security": {
+      "transport_encryption": {
+        "enabled": true
+      }
+    }
+  },
+  "elasticsearch": {
+    "xpack_enabled": true
+  }
+}
+EOF
+```
+
+```
+dcos beta-elastic --name=$your_service_name update start --options=elastic_with_tls.json
+```
+
+## Now you can only access your Elasticsearch cluster through HTTPS
+
+```bash
+dcos task exec -it coordinator-0-node curl -k -u elastic:changeme https://coordinator.$your_service_name.l4lb.thisdcos.directory:9200 
```

Comment:
Precede the `bash` command with a `$`, and don't use the `$your_service_name` variable unless it is explicitly specified.

Reply:
This is also a bit misleading as we are not accessing the cluster here. We are just running a command on the node directly. It may be more useful to link to information on how one should connect to a TLS-enabled Elastic cluster.